### PR TITLE
helm chart for load balancer

### PR DESCRIPTION
**Pull request by ramaraosrikakulapu**

### CODE REVIEW

File: k8s/example/Chart.yaml [hunk 1]
```diff
@@ -33,3 +33,6 @@ dependencies:
 - name: agent+plg
   version: <AGENT_PLG_CHART_REV>
   repository: "file://./../agent+plg"
+- name: agent+lber
```
Comment (ayasuda-ge) [inline k8s/example/Chart.yaml]: k8s/example should only be used for agent chart.

File: k8s/example/values.yaml [hunk 2]
```diff
@@ -73,52 +73,56 @@ global:
...
-        cpu: 100m
-        memory: 128Mi
+        cpu: 200m
+        memory: 512Mi
       requests:
         cpu: 100m
         memory: 128Mi
-    svcPortNum: 18080
-    svcHealthPortNum: 18081
+    svcPortNum: 8080
```
Comment (ayasuda-ge) [inline k8s/example/values.yaml]: k8s/example should only be used for agent chart.

File: k8s/agent+helper/templates/_agent.tpl [hunk 3]
```diff
@@ -0,0 +1,178 @@
+{{/* vim: set filetype=mustache: */}}
```
Comment (ayasuda-ge) [inline k8s/agent+helper/templates/_agent.tpl]: duplicated signature. Suggestion for the removal of this file.

**Review by ayasuda-ge — CHANGES_REQUESTED** [review]
please see comments.

File: scripts/k8s-deployment-sim.sh [hunk 4]
```diff
@@ -35,7 +35,7 @@ printf "\n\n\n*** [3] install client with remote vln template in minikube\n"
 yq e '.global.agtK8Config.withPlugins.tls.enabled = false' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.enabled = true' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.remote = true' -i k8s/example/values.yaml
-helm install k8s/example --set-file global.agtConfig=k8s/example/client+vln.env --generate-name
+helm install k8s/example --set-file global.agtConfig=k8s/example/client+vln.env
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: leave the original working as-is.

File: scripts/spec-sim.sh [hunk 5]
```diff
@@ -40,33 +43,37 @@ eval "sed -i -e 's#{{EC_TEST_SST}}#${EC_TEST_SST}#g' k8s/example/gateway.env"
...
 
 printf "update dependencies in example chart for test"
 helm dependency update k8s/example
+ls -la k8s/example/charts/
 
 printf "\n\n\n*** test server with tls template\n"
 yq e '.global.agtK8Config.withPlugins.tls.enabled = true' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.enabled = false' -i k8s/example/values.yaml
-helm template k8s/example --debug --set-file global.agtConfig=k8s/example/server+tls.env
+helm template my-app k8s/example --debug --set-file global.agtConfig=k8s/example/server+tls.env
```
Comment (ayasuda-ge) [inline scripts/spec-sim.sh]: leave the example working as-is.

File: scripts/spec-sim.sh [hunk 6]
```diff
@@ -40,33 +43,37 @@ eval "sed -i -e 's#{{EC_TEST_SST}}#${EC_TEST_SST}#g' k8s/example/gateway.env"
...
 
 printf "\n\n\n*** test client with local vln template\n"
 yq e '.global.agtK8Config.withPlugins.tls.enabled = false' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.enabled = true' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.remote = false' -i k8s/example/values.yaml
-helm template k8s/example --debug --set-file global.agtConfig=k8s/example/client+vln.env
+helm template my-app k8s/example --debug --set-file global.agtConfig=k8s/example/client+vln.env
```
Comment (ayasuda-ge) [inline scripts/spec-sim.sh]: leave the example working as-is.

File: scripts/spec-sim.sh [hunk 7]
```diff
@@ -40,33 +43,37 @@ eval "sed -i -e 's#{{EC_TEST_SST}}#${EC_TEST_SST}#g' k8s/example/gateway.env"
...
 
 printf "\n\n\n*** test client with remote vln template\n"
 yq e '.global.agtK8Config.withPlugins.tls.enabled = false' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.enabled = true' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.remote = true' -i k8s/example/values.yaml
-helm template k8s/example --debug --set-file global.agtConfig=k8s/example/client+vln.env
+helm template my-app k8s/example --debug --set-file global.agtConfig=k8s/example/client+vln.env
```
Comment (ayasuda-ge) [inline scripts/spec-sim.sh]: leave the example working as-is.

File: scripts/k8s-deployment-sim.sh [hunk 8]
```diff
@@ -1,5 +1,20 @@
 #!/bin/bash
 
+printf "\n\n\n*** [0] install server with tls template and HA in k8s\n"
+yq e '.global.agtK8Config.withPlugins.tls.enabled = true' -i k8s/example/values.yaml
+yq e '.global.agtK8Config.withPlugins.vln.enabled = false' -i k8s/example/values.yaml
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: use k8s/agent+lber/value.yaml instead.

File: scripts/k8s-deployment-sim.sh [hunk 9]
```diff
@@ -1,5 +1,20 @@
...
+helm install k8s/agent+lber --set-file global.agtConfig=k8s/example/server+tls.env --generate-name
+printf "\n\n\n*** [0.1] verify installation\n"
+kubectl get deployments && kubectl get pods && kubectl get services
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: output ingress checking for lber.

File: scripts/k8s-deployment-sim.sh [hunk 10]
```diff
@@ -1,5 +1,20 @@
...
+printf "\n\n\n*** [0.2] verify deployment spec\n"
+kubectl describe deployments $(kubectl get deployments|grep agent-plg|awk '{print $1}'|head -n 1)
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: its should be a "lber" deployment instead.

File: scripts/k8s-deployment-sim.sh [hunk 11]
```diff
@@ -1,5 +1,20 @@
...
+printf "\n\n\n*** [0.3] verify service spec\n"
+kubectl describe services $(kubectl get services|grep agent-plg|awk '{print $1}'|head -n 1)
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: "lber" service + "agent" service.

File: scripts/k8s-deployment-sim.sh [hunk 12]
```diff
@@ -1,5 +1,20 @@
...
+#printf "\n\n\n*** done debug go ahead delete all.\n\n"
+printf "\n\n\n*** [0.4] clear installation\n"
+kubectl delete --all deployments && kubectl delete --all pods && kubectl delete --all services
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: remove ingress as last step.

File: scripts/spec-sim.sh [hunk 13]
```diff
@@ -40,54 +43,63 @@ eval "sed -i -e 's#{{EC_TEST_SST}}#${EC_TEST_SST}#g' k8s/example/gateway.env"
...
 printf "\n\n\n*** test client with remote vln template\n"
 yq e '.global.agtK8Config.withPlugins.tls.enabled = false' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.enabled = true' -i k8s/example/values.yaml
 yq e '.global.agtK8Config.withPlugins.vln.remote = true' -i k8s/example/values.yaml
-helm template k8s/example --debug --set-file global.agtConfig=k8s/example/client+vln.env
+#helm template k8s/example --debug --set-file global.agtConfig=k8s/example/client+vln.env
 
 printf "\n\n\n*** test gateway agt template\n"
-helm template k8s/example --debug --set-file global.agtConfig=k8s/example/gateway.env
+#helm template k8s/example --debug --set-file global.agtConfig=k8s/example/gateway.env
```
Comment (ayasuda-ge) [inline scripts/spec-sim.sh]: ensure to uncomment.

File: README.md [hunk 14]
```diff
@@ -25,21 +25,24 @@ The diagram illustrates the usage of the connectivity model in k8s
 #### Add Dependency Repo
 ```bash
 # bootstrap a chart
-$ helm create example
+$ helm create example -n namespace
 
 # REQUIRED: add the helper library as the dependency to support usage
-$ helm repo add agent+helper https://ec-release.github.io/oci/agent+helper/<version. E.g. "0.1.0">
+$ helm repo add agent+helper https://ec-release.github.io/oci/agent+helper/<version. E.g. "0.1.0"> -n namespace
```
Comment (ayasuda-ge) [inline README.md]: pls replace "..ec-release.github.io.." to the following-

https://raw.githubusercontent.com/EC-Release/helmcharts/disty/agent+lber/<version>

and refactoring other links as well.

File: k8s/example/Chart.yaml (hunk 1 above)
Comment (ramaraosrikakulapu) [reply k8s/example/Chart.yaml]: agent+lber dependency removed

File: k8s/example/values.yaml (hunk 2 above)
Comment (ramaraosrikakulapu) [reply k8s/example/values.yaml]: All changes reverted for example/values.yaml as it is used only for agent test

File: k8s/agent+helper/templates/_agent.tpl (hunk 3 above)
Comment (ramaraosrikakulapu) [reply k8s/agent+helper/templates/_agent.tpl]: All functions in _agent.tpl separated and deleted the duplicate ones

File: scripts/k8s-deployment-sim.sh (hunk 4 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Agent test cases reverted to as-is

File: scripts/spec-sim.sh (hunk 5 above)
Comment (ramaraosrikakulapu) [reply scripts/spec-sim.sh]: Agent test cases reverted to as-is

File: scripts/spec-sim.sh (hunk 6 above)
Comment (ramaraosrikakulapu) [reply scripts/spec-sim.sh]: Agent test cases reverted to as-is

File: scripts/spec-sim.sh (hunk 7 above)
Comment (ramaraosrikakulapu) [reply scripts/spec-sim.sh]: Agent test cases reverted to as-is

File: scripts/k8s-deployment-sim.sh (hunk 8 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Now lber tested with agent+lber/values.yaml and example/values.yaml used only for agent test.

File: scripts/k8s-deployment-sim.sh (hunk 9 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Added ingress to get the details

File: scripts/k8s-deployment-sim.sh (hunk 10 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Updated to display deployment with name 'agentlber' and services 'agent' and 'agentlber'

File: scripts/k8s-deployment-sim.sh (hunk 11 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Updated to display deployment with name 'agentlber' and services 'agent' and 'agentlber'

File: scripts/k8s-deployment-sim.sh (hunk 12 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Added to ingress to delete

File: scripts/spec-sim.sh (hunk 13 above)
Comment (ramaraosrikakulapu) [reply scripts/spec-sim.sh]: Agent tests uncommented

File: README.md (hunk 14 above)
Comment (ramaraosrikakulapu) [reply README.md]: README updated with new URL's

File: README.md [hunk 15]
```diff
@@ -59,43 +62,78 @@ dependencies:
 - name: agent+plg
   version: 0.1.1
   repository: "@agent+plg"
+# OPTIONAL
+- name: agentlber
```
Comment (ayasuda-ge) [inline README.md]: *agent+lber

File: k8s/agent+helper/templates/_common.tpl [hunk 16]
```diff
@@ -36,3 +36,15 @@ true
 {{- end -}}
 {{- end -}}
 {{- end -}}
+
+
+{{/*
```
Comment (ayasuda-ge) [inline k8s/agent+helper/templates/_common.tpl]: deprecated. please remove if there is no use in lber.

File: k8s/example/values.yaml [hunk 17]
```diff
@@ -73,8 +73,8 @@ global:
     # PCS CF1/CF3 Cluster, whereas some can simply ignore the usage.
     resources:
       limits:
-        cpu: 100m
-        memory: 128Mi
+        cpu: 200m
```
Comment (ayasuda-ge) [inline k8s/example/values.yaml]: suggestion to leave the config set as-is solely for agent

File: k8s/example/values.yaml [hunk 18]
```diff
@@ -84,10 +84,10 @@ global:
     releaseTag: v1
     # A download-able url for an "decrypted" agent binary that will be used as
     # the connectivity runtime in the target environment
-    binaryURL: https://github.com/EC-Release/sdk/releases/agent
+    # binaryURL: https://github.com/EC-Release/sdk/releases/agent
```
Comment (ayasuda-ge) [inline k8s/example/values.yaml]: binaryURL is currently in production usage for v213. Suggest to uncomment and leave as-is.

File: scripts/k8s-deployment-sim.sh [hunk 19]
```diff
@@ -1,5 +1,20 @@
 #!/bin/bash
 
+printf "\n\n\n*** [0] install server with tls template and HA in k8s\n"
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: suggestion "..install HA gateway template in k8s.."

File: scripts/k8s-deployment-sim.sh [hunk 20]
```diff
@@ -1,5 +1,20 @@
...
+yq e '.global.agtK8Config.withPlugins.tls.enabled = true' -i k8s/agent+lber/values.yaml
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: redundant per discussion

File: scripts/k8s-deployment-sim.sh [hunk 21]
```diff
@@ -1,5 +1,20 @@
...
+yq e '.global.agtK8Config.withPlugins.vln.enabled = false' -i k8s/agent+lber/values.yaml
+helm install my-app k8s/agent+lber --set-file global.agtConfig=k8s/example/server+tls.env
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: update to "gateway.env"

File: scripts/k8s-deployment-sim.sh [hunk 22]
```diff
@@ -1,5 +1,20 @@
...
+printf "\n\n\n*** [0.1] verify installation\n"
+kubectl get deployments && kubectl get pods && kubectl get services && kubectl get ingress
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: add sts checking.

Test requirement:
igs x1
svc x2
pod xlber-replicaCount
pod xagt-replicaCount
deployment x1
sts x1

File: scripts/k8s-deployment-sim.sh [hunk 23]
```diff
@@ -1,5 +1,20 @@
...
+printf "\n\n\n*** [0.2] verify deployment spec\n"
+kubectl describe deployments $(kubectl get deployments|grep agentlber|awk '{print $1}'|head -n 1)
+printf "\n\n\n*** [0.3] verify service spec\n"
+kubectl describe services $(kubectl get services|grep agent|awk '{print $1}'|head -n 1)
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: .. | grep lber | ..

File: scripts/k8s-deployment-sim.sh [hunk 24]
```diff
@@ -1,5 +1,20 @@
...
+#printf "\n\n\n*** done debug go ahead delete all.\n\n"
+printf "\n\n\n*** [0.4] clear installation\n"
+kubectl delete --all deployments && kubectl delete --all pods && kubectl delete --all services && kubectl delete --all ingress
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh]: + delete sts

**Review by ayasuda-ge — CHANGES_REQUESTED** [review]
please consider the following update

File: scripts/k8s-deployment-sim.sh (hunk 24 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: done

File: scripts/k8s-deployment-sim.sh (hunk 23 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: done

File: k8s/example/Chart.yaml (hunk 1 above)
Comment (ayasuda-ge) [reply k8s/example/Chart.yaml]: this may fail for lber. Please remove

File: scripts/k8s-deployment-sim.sh (hunk 22 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: All components are showed in logs

File: scripts/k8s-deployment-sim.sh (hunk 21 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: done

File: scripts/k8s-deployment-sim.sh (hunk 20 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: Running with gateway.env. But still wants to disable plugins.

File: scripts/k8s-deployment-sim.sh (hunk 19 above)
Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh]: updated

File: k8s/example/values.yaml (hunk 18 above)
Comment (ramaraosrikakulapu) [reply k8s/example/values.yaml]: reverted

File: k8s/example/values.yaml (hunk 17 above)
Comment (ramaraosrikakulapu) [reply k8s/example/values.yaml]: reverted

File: k8s/agent+helper/templates/_common.tpl (hunk 16 above)
Comment (ramaraosrikakulapu) [reply k8s/agent+helper/templates/_common.tpl]: removed

File: README.md (hunk 15 above)
Comment (ramaraosrikakulapu) [reply README.md]: updated

File: scripts/k8s-deployment-sim.sh [hunk 25]
```diff
@@ -1,5 +1,23 @@
 #!/bin/bash
 
+printf "\n\n\n*** [0] install gateway with HA in k8s\n"
+yq e '.global.agtK8Config.withPlugins.tls.enabled = false' -i k8s/agent+lber/values.yaml
+yq e '.global.agtK8Config.withPlugins.vln.enabled = false' -i k8s/agent+lber/values.yaml
+helm install my-app k8s/agent+lber --set-file global.agtConfig=k8s/example/gateway.env
+printf "\n\n\n*** [0.1] verify installation\n"
+kubectl get deployments && kubectl get sts && kubectl get pods && kubectl get services && kubectl get ingress
+printf "\n\n\n*** [0.2] verify deployment spec\n"
```
Comment (ayasuda-ge) [inline scripts/k8s-deployment-sim.sh:9]: add ingress spec checking to benefit future debugging.

Comment (ramaraosrikakulapu) [reply scripts/k8s-deployment-sim.sh:9]: Update the script

**Pull request merged**